packages/core/cyclist.mjs

```diff
@@ -52,6 +52,10 @@ export class Cyclist {
               // see https://github.com/tidalcycles/strudel/pull/1004
               const deadline = targetTime - phase;
               onTrigger?.(hap, deadline, duration, this.cps, targetTime);
+              if (hap.value.cps !== undefined && this.cps != hap.value.cps) {
+                this.cps = hap.value.cps;
+                this.num_ticks_since_cps_change = 0;
```

Comment:
I suppose this shouldn't be 0 as we're partway through a tick. Perhaps somewhere between -1 and 0 ??

Reply:
It's an integer counter...  the tick is the smallest unit of handled time. In a tick you feed a fixed-size audio buffer. You could go down to the single sample but i think performance could be killed.